app/models/friendship.rb

```diff
@@ -1,11 +1,55 @@
+class SelfFriendshipValidator < ActiveModel::Validator
+  def validate(record)
+    record.errors[:base].push("You can't be friends with yourself") if record.user_id == record.friend_id
```

Comment:
- [ ] [OPTIONAL]: This should be a mandatory requirement because one of the past CR requested it but the last CR didn't check it and request it.
- You did handle the case of not being able to send a friend request to yourself, however, a better solution would be to not even render the send friendship button at all, you can check the code that I used to achieve this in my project, let me know if you have any question about it but basically, the first thing checked is if the friend.id is equal to the current_user.id, if true, we return and exit the function, otherwise, we render the request friendship button accordingly, this is how you can create smart buttons that appear or not depending on who the user is.

- [How I handled this issue](https://github.com/e71az/social-media/blob/development/app/helpers/friendship_helper.rb)

Reply:
![Screenshot from 2021-09-22 11-32-14](https://user-images.githubusercontent.com/65048528/134385402-a2cbb46f-3142-424e-a364-3a6ad597ea27.png)

I would appreciate if you could double-check on this, there is an alert that says 'you can't be friends with yourself.' I implemented this for my previous pull request, as you can see in the image.
Thanks 

I'll be adding logic to the next request

Reply:
Hello Angela, 

I did mention that you handled the case and I also tested it and saw that notification. I just suggested that a better solution would be not to show the button to add yourself 👍

Reply:
Thanks Elias, I was already overwhelmed by all the changes. The project was a little hard and the time is stressing me, sorry I didn't read correctly.:)